lib/net/ssh/transport/packet_stream.rb

```diff
@@ -161,13 +161,28 @@ def enqueue_packet(payload)
             mac = client.hmac.digest([client.sequence_number, mac_data].pack("NA*"))
 
             message = mac_data + mac
+          elsif client.hmac.aead
+            # Details of this implementation can be found in RFC 5647
+            unencrypted_data = [padding_length, payload, padding].pack("CA*A*")
+
+            # Despite SSH spec, when using AEAD encryption, packet_length is not ciphered. It is transported as identity,
+            # and added in the auth_data. See RFC 5647 7.3
+            client.cipher.auth_data = [packet_length].pack('N')
+            encrypted_data = client.update_cipher(unencrypted_data)
+            encrypted_data << client.final_cipher
+
+            # For SSH, GCM auth_tag acts as MAC
+            mac = client.auth_tag
+
+            message = [packet_length].pack("N") + encrypted_data + [mac].pack("A*")
```

Comment:
I'm refactoring this part in chacha20-poly1035, and it looks like this:
```ruby
if client.cipher.implicit_mac?
   unencrypted_data = [padding_length, payload, padding].pack("CA*A*")
   message = client.cipher.update_cipher_mac(unencrypted_data, client.sequence_number)
```

I think this would work for this use case as well, what do you think?